### PR TITLE
v4.1.x: MPI_Init_thread(3) man page updates

### PR DESCRIPTION
**Pull request by jsquyres**
4.1.x RMs: there are 2 commits on this PR:

1. A cherry pick from master (from #8399)
1. A cherry pick from v4.0.x (from #8400), because it fixes a statement about C++ that no longer exists on master

Thanks to Andreas Lösel for bringing the issue to our attention.

### CODE REVIEW

**Review by bosilca — APPROVED** [review]

**Comment by jsquyres** [comment]
@rajachan This includes a cherry pick from v4.0.x; don't merge here until #8400 merges.

**Pull request merged**
